cypress/e2e/pages/topicPage/tests.js

```diff
@@ -31,69 +31,74 @@ export default ({ service, pageType, variant }) => {
             otherVariant = variant === 'simp' ? 'trad' : 'simp';
           }
         }
-
         // Gets the topic page data for all the tests
-        cy.getPageData({
-          service,
-          pageType: 'topic',
-          variant,
-        }).then(({ body }) => {
-          topicTitle = body.data.title;
-          variantTopicId = body.data.variantTopicId;
-          pageCount = body.data.pageCount;
-          numberOfItems = body.data.curations[0].summaries.length;
-          firstItemHeadline = body.data.curations[0].summaries[0].title;
-          messageBanner = body.data.curations.find(
+        cy.getPageDataFromWindow().then(data => {
+          const { pageData } = data;
+          topicTitle = pageData.title;
+          variantTopicId = pageData.scriptSwitchId;
+          pageCount = pageData.pageCount;
+          numberOfItems = pageData.curations?.[0]?.summaries.length;
+          firstItemHeadline = pageData.curations?.[0]?.summaries?.[0]?.title;
+          messageBanner = pageData.curations?.find(
             curation =>
               curation.visualProminence === 'NORMAL' &&
               curation.visualStyle === 'BANNER',
           );
+          // const pageData = data.pageData
         });
         cy.log(`topic id ${topicId}`);
       }
     });
 
     describe(`Page content`, () => {
+      beforeEach(() => {
+        // make sure we always start from the path being tested to make the tests deterministic and not reliant on order
+        // as otherwise some tests can change the path and affect subsequent tests (i.e. when you change page script)
+        cy.visit(currentPath);
```

Comment:
@karinathomas instead of using `cy.go('back')` I added `cy.visit(currentPath)` in a `beforeEach`. I think this is more stable, as go back in specific senarios could fail if we have more than one click in a test or if we change the order of the tests.

Reply:
I tried moving this to the top `describe` but it breaks some tests - I am not totally sure of the reasons tbh, maybe some of the other tests rely on the base service link and adding it to the top describe conflicts with the `cy.visit` in `before` the top test runner [here](https://github.com/bbc/simorgh/blob/8ebb8e4091ec93d09f7a25b619eda076f9a04ae4/cypress/support/helpers/runTestsForPage.js#L64-L65).

Is it ok to leave it as it is, or do you think I should investigate more why it doesn't work from the top level?

Reply:
I think @LilyL0u has figured this out: https://github.com/bbc/simorgh/pull/11554#issuecomment-2078976376. Let's leave it where it is. Thanks!